### PR TITLE
Update Azuresql doc

### PR DESCRIPTION
**Pull request by MilanBariya**
All the below issue is Fixd for Azuresql

- [x] Move the "workflow deployment error" out of the configuration steps and just as a troubleshooting section at the end
- [x] On the YAML side: Connection Options and Connection argument should be at the bottom with their own specific example, as we are not using them in the example YAML.
- [x] In the workflow auth, only keep JWT. Have a separate page for the rest of the SSO workflow config auth and add a link to that page from the connector. We'll be deprecating that, so we might as well remove it from the main flow here.
- [x] When talking about JWT tokens, add a link to the page explaining how to obtain them.
- [x] On the DAG examples (ingestion, profiler,...), use the same component as with the YAML: explain every block
- [x] The source configuration in the profiler YAML should also have relevant examples and information
- [x] In the profiler, lineage, usage,... YAMLs we don't need the serviceConnection, only the serviceName. Clean the YAMLs and explain the behavior (since we already have the metadata ingested, the ingestion bot can get the credentials from the server  etc)

### CODE REVIEW

File: content/v0.13.2/connectors/database/azuresql/airflow.md [hunk 1]
```diff
@@ -134,25 +106,111 @@ The `sourceConfig` is defined [here](https://github.com/open-metadata/OpenMetada
...
+      password: password
+```
+```yaml {% srNumber=3 %}
+      hostPort: hostPort
+```
+```yaml {% srNumber=4 %}
+      database: database_name
+```
+```yaml {% srNumber=5 %}
+     # driver: ODBC Driver 18 for SQL Server (default)
```
Comment (pmbrull) [inline content/v0.13.2/connectors/database/azuresql/airflow.md:166]: This is not aligned with the line above

File: content/v0.13.2/connectors/database/azuresql/airflow.md [hunk 2]
```diff
@@ -134,25 +106,111 @@ The `sourceConfig` is defined [here](https://github.com/open-metadata/OpenMetada
...
+```
+```yaml {% srNumber=6 %}
+      # connectionOptions:
+        # key: Value
```
Comment (pmbrull) [inline content/v0.13.2/connectors/database/azuresql/airflow.md:170]: use `key: value` in lowercase

File: content/v0.13.2/connectors/database/azuresql/airflow.md [hunk 3]
```diff
@@ -134,25 +106,111 @@ The `sourceConfig` is defined [here](https://github.com/open-metadata/OpenMetada
...
+```
+```yaml {% srNumber=7 %}
+      # connectionArguments:
+        # authenticator: externalbrowser
```
Comment (pmbrull) [inline content/v0.13.2/connectors/database/azuresql/airflow.md:174]: don't show the `authenticator` here. That's just an example (which we should specify that only works from the CLI). We can stick to `key: value` as well

File: content/v0.13.2/connectors/database/azuresql/airflow.md [hunk 4]
```diff
@@ -176,127 +238,76 @@ workflowConfig:
     securityConfig:
       jwtToken: "{bot_jwt_token}"
 ```
+- To enable JWT Tokens, you will get more details [here](/deployment/security/enable-jwt-tokens).
```
Comment (pmbrull) [inline content/v0.13.2/connectors/database/azuresql/airflow.md:241]: Thanks for organizing the links, but I'd improve the wording a bit:

```
## OpenMetadata JWT Auth

JWT tokens will allow your clients to authenticate against the OpenMetadata server. To enable JWT Tokens, you will get more details here(link).

<Show code>

You can refer to the JWT Troubleshooting section (link) for any issues in your JWT configuration. If you need information on configuring the ingestion with other security providers in your bots, you can follow this doc (link).
```

File: content/v0.13.2/connectors/database/azuresql/airflow.md [hunk 5]
```diff
@@ -176,127 +238,76 @@ workflowConfig:
...
-  openMetadataServerConfig:
-    hostPort: "http://localhost:8585/api"
-    authProvider: google
-    securityConfig:
-      secretKey: "{path-to-json-creds}"
-```
+{% codeInfo srNumber=11 %}
 
-### Okta SSO
+- **yaml**: A Python package that can load YAML files and convert them into Python objects.
```
Comment (pmbrull) [inline content/v0.13.2/connectors/database/azuresql/airflow.md:260]: I would not drill down to every specific dependency, but just give an overview here.

The important part is to explain the `metadata` related functions/classes. The rest we can just say something like "we are importing all the basic requirements to parse YAMLs, handle dates and build our DAG"

File: content/v0.13.2/connectors/database/azuresql/airflow.md [hunk 6]
```diff
@@ -176,127 +238,76 @@ workflowConfig:
...
-    securityConfig:
-      clientId: "{your_client_id}"
-      secretKey: "{your_client_secret}"
-      domain: "{your_domain}"
-```
+{% /codeInfo %}
 
-### KeyCloak SSO
+{% codeInfo srNumber=12 %}
+Default arguments for all tasks in the Airflow DAG.
```
Comment (pmbrull) [inline content/v0.13.2/connectors/database/azuresql/airflow.md:277]: The code here might need some spacing

File: content/v0.13.2/connectors/database/azuresql/airflow.md [hunk 7]
```diff
@@ -176,127 +238,76 @@ workflowConfig:
...
 
-### 2. Prepare the Ingestion DAG
+- **"retry_delay": timedelta(minutes=5)"**: Specifies the delay between task retries. In this case, the delay is set to 5 minutes.
 
-Create a Python file in your Airflow DAGs directory with the following contents:
+- **"execution_timeout": timedelta(minutes=60)"**: Specifies the maximum duration of time that a task should run for. If a task exceeds this duration, it will be marked as failed.
+{% /codeInfo %}
+
+{% codeInfo srNumber=13 %}
+- **config**: Specifies config for the profiler as we prepare above.
```
Comment (pmbrull) [inline content/v0.13.2/connectors/database/azuresql/airflow.md:291]: The first DAG is for the metadata ingestion. The second for the profiler.

File: content/v0.13.2/connectors/database/azuresql/airflow.md [hunk 8]
```diff
@@ -176,127 +238,76 @@ workflowConfig:
...
+{% /codeInfo %}
+
+{% codeInfo srNumber=14 %}
+- **metadata_ingestion_workflow()**: This code defines a function `metadata_ingestion_workflow()` that loads a YAML configuration, creates a `Workflow` object, executes the workflow, checks its status, prints the status to the console, and stops the workflow.
+{% /codeInfo %}
+
+{% codeInfo srNumber=15 %}
+- **DAG**:  The DAG is created using the default arguments defined in the `default_args` dictionary. The DAG is set to start one day ago, is not paused upon creation, runs every 5 minutes using the cron syntax '*/5 * * * *', and does not catch up on any missed runs.
```
[the remark this answers is not in the document]
Comment (pmbrull) [reply content/v0.13.2/connectors/database/azuresql/airflow.md:299]: We can also say thay they can tune the DAG configurations to whatever fits their requirements

File: content/v0.13.2/connectors/database/azuresql/airflow.md [hunk 9]
```diff
@@ -347,136 +366,261 @@ with DAG(
...
+{% codePreview %}
+
+{% codeInfoContainer %}
+
+{% codeInfo srNumber=16 %}
+**generateSampleData**: Option to turn on/off generating sample data.
+{% /codeInfo %}
+
+{% codeInfo srNumber=17 %}
+**profileSample**: Percentage of data or no. of rows we want to execute the profiler and tests on.
```
Comment (pmbrull) [inline content/v0.13.2/connectors/database/azuresql/airflow.md:394]: In these sections, add any links to other docs (such as profiler docs, metrics, etc) to make it easier for users to navigate

File: content/v0.13.2/connectors/database/azuresql/airflow.md (hunk 4 above)
Comment (MilanBariya) [reply content/v0.13.2/connectors/database/azuresql/airflow.md]: Done

File: content/v0.13.2/connectors/database/azuresql/airflow.md (hunk 5 above)
Comment (MilanBariya) [reply content/v0.13.2/connectors/database/azuresql/airflow.md]: Done

File: content/v0.13.2/connectors/database/azuresql/airflow.md (hunk 6 above)
Comment (MilanBariya) [reply content/v0.13.2/connectors/database/azuresql/airflow.md]: Done

File: content/v0.13.2/connectors/database/azuresql/airflow.md (hunk 9 above)
Comment (MilanBariya) [reply content/v0.13.2/connectors/database/azuresql/airflow.md:378]: Done

File: content/v0.13.2/connectors/database/azuresql/airflow.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (MilanBariya) [reply content/v0.13.2/connectors/database/azuresql/airflow.md]: Done

File: content/v0.13.2/connectors/database/azuresql/airflow.md (hunk 7 above)
Comment (MilanBariya) [reply content/v0.13.2/connectors/database/azuresql/airflow.md]: Done

**Review by pmbrull — COMMENTED** [review]
Thanks, some comments:

- In the DAG imports, we said to skip the "common Python" ones, but we want to explain the ones related to OpenMetadata
- No need to explain all the default arguments in the DAG, just an overview
- The DAG file still has no spacing, it's hard to read

Thanks

**Review by pmbrull — APPROVED** [review]

**Pull request merged**